src/api/layer.js

```diff
@@ -179,18 +127,15 @@ export default class Layer {
      * @instance
      * @api
      */
-    setStyle(style) {
+    async setStyle(style) {
         this._checkStyle(style);
-        return this._styleChanged(style).then(r => {
-            if (this._style) {
-                this._style.onChange(null);
-            }
-            this._style = style;
-            this._style.onChange(() => {
-                this._styleChanged(style);
-            });
-            return r;
-        });
+
+        await this._styleChanged(style);
+        if (this._style) {
+            this._style.onChange(null);
+        }
+        this._style = style;
+        this._style.onChange(this._styleChanged.bind(this));
```

Comment:
`_styleChanged` doesn't receive a style parameter?

Reply:
yes, the `style` object will call this method passing `this` as parameter

Reply:
Ok